File: android-native/app/src/main/kotlin/com/wheelword/game/ui/CoinInfoDialog.kt
package com.wheelword.game.ui

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wheelword.game.HINT_COIN_COST
import com.wheelword.game.theme.BalooFamily
import com.wheelword.game.theme.GameColors
import com.wheelword.game.theme.NunitoFamily

/**
 * Modal explaining the coin economy. Triggered from the coin pill in
 * the HUD. The "Buy a hint for [HINT_COIN_COST] points" line is the
 * point of the dialog — players who run out of free hints need to know
 * they can keep advancing by spending coins.
 */
@Composable
fun CoinInfoDialog(coins: Int, onDismiss: () -> Unit) {
    ModalShell(onDismiss = onDismiss, widthFraction = 0.88f) {
        StarCoinIcon(size = 56.dp)
        Spacer(Modifier.height(10.dp))
        Text(
            text = "Points",
            color = Color.White,
            fontFamily = BalooFamily,
            fontSize = 24.sp,
            fontWeight = FontWeight.ExtraBold,
        )
        Spacer(Modifier.height(2.dp))
        Text(
            text = "You have $coins",
            color = GameColors.InkSoft,
            fontFamily = NunitoFamily,
            fontWeight = FontWeight.Bold,
            fontSize = 14.sp,
        )
        Spacer(Modifier.height(16.dp))

        CoinInfoRow(
            glyph = "✏️",
            title = "Find words",
            detail = "+2 points per word — grid answers and bonus words.",
        )
        CoinInfoRow(
            glyph = "🏆",
            title = "Finish a level",
            detail = "+10 bonus when you complete the crossword.",
        )
        CoinInfoRow(
            glyph = "🎁",
            title = "Daily spin",
            detail = "Spin the wheel once a day for a coin / hint reward.",
        )
        CoinInfoRow(
            glyph = "💡",
            title = "Buy a hint for $HINT_COIN_COST",
            detail = "When your free hints are gone, tap the lightbulb to spend $HINT_COIN_COST points and reveal a letter.",
        )

        Spacer(Modifier.height(18.dp))
        CandyButton(
            text = "Got it",
            onClick = onDismiss,
            color = GameColors.Blue,
            colorDeep = GameColors.BlueDeep,
            fontSize = 17.sp,
            modifier = Modifier.fillMaxWidth(),
            verticalPadding = 11.dp,
        )
    }
}

@Composable
private fun CoinInfoRow(glyph: String, title: String, detail: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 5.dp),
        verticalAlignment = Alignment.Top,
    ) {
        Text(
            text = glyph,
            fontSize = 20.sp,
            modifier = Modifier
                .width(32.dp)
                .padding(top = 1.dp),
        )
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = title,
                color = Color.White,
                fontFamily = BalooFamily,
                fontSize = 15.sp,
                fontWeight = FontWeight.Bold,
            )
            Text(
                text = detail,
                color = GameColors.InkSoft,
                fontFamily = NunitoFamily,
                fontWeight = FontWeight.SemiBold,
                fontSize = 12.sp,
            )
        }
    }
}

File: ios-native/WordWheel/Views/CoinInfoDialog.swift
import SwiftUI

/// Modal explaining the coin economy. Triggered from the coin badge in
/// `TopBar`. The "Buy a hint for HINT_COIN_COST points" line is the
/// point of the dialog — players who run out of free hints need to
/// know they can keep advancing by spending coins.
struct CoinInfoDialog: View {
    let coins: Int
    let onDismiss: () -> Void

    var body: some View {
        ZStack {
            Color.black.opacity(0.78)
                .ignoresSafeArea()
                .onTapGesture(perform: onDismiss)

            VStack(alignment: .center, spacing: 0) {
                ZStack {
                    Circle()
                        .fill(LinearGradient(
                            colors: [Color(hex: 0xFFFFE070), Color(hex: 0xFFFFB020)],
                            startPoint: .topLeading, endPoint: .bottomTrailing,
                        ))
                        .overlay(Circle().stroke(Color(hex: 0xFFB37400), lineWidth: 2))
                    Text("★")
                        .font(.system(size: 34, weight: .black))
                        .foregroundColor(.white)
                }
                .frame(width: 56, height: 56)
                .padding(.top, 24)

                Text("Points")
                    .font(.system(size: 22, weight: .bold))
                    .foregroundColor(.white)
                    .padding(.top, 10)

                Text("You have \(coins)")
                    .font(.system(size: 14))
                    .foregroundColor(.white.opacity(0.75))
                    .padding(.top, 2)

                VStack(alignment: .leading, spacing: 5) {
                    CoinInfoRow(glyph: "✏️", title: "Find words",
                                detail: "+2 points per word — grid answers and bonus words.")
                    CoinInfoRow(glyph: "🏆", title: "Finish a level",
                                detail: "+10 bonus when you complete the crossword.")
                    CoinInfoRow(glyph: "🎁", title: "Daily spin",
                                detail: "Spin the wheel once a day for a coin / hint reward.")
                    CoinInfoRow(glyph: "💡",
                                title: "Buy a hint for \(HINT_COIN_COST)",
                                detail: "When your free hints are gone, tap the lightbulb to spend \(HINT_COIN_COST) points and reveal a letter.")
                }
                .padding(.horizontal, 20)
                .padding(.top, 18)

                Button(action: onDismiss) {
                    Text("Got it")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.white)
                        .padding(.horizontal, 36).padding(.vertical, 12)
                        .background(RoundedRectangle(cornerRadius: 24).fill(GameColors.submitBg))
                }
                .buttonStyle(.plain)
                .padding(.vertical, 22)
            }
            .frame(maxWidth: 420)
            .padding(.horizontal, 12)
            .background(
                RoundedRectangle(cornerRadius: 20).fill(GameColors.completeBg)
            )
            .shadow(color: .black.opacity(0.5), radius: 18)
            .padding(.horizontal, 24)
        }
    }
}

private struct CoinInfoRow: View {
    let glyph: String
    let title: String
    let detail: String

    var body: some View {
        HStack(alignment: .top, spacing: 10) {
            Text(glyph)
                .font(.system(size: 20))
                .frame(width: 30, alignment: .center)
                .padding(.top, 1)
            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(.system(size: 15, weight: .semibold))
                    .foregroundColor(.white)
                Text(detail)
                    .font(.system(size: 12))
                    .foregroundColor(.white.opacity(0.72))
                    .fixedSize(horizontal: false, vertical: true)
            }
        }
        .padding(.vertical, 4)
    }
}
